src/shims/posix/fs.rs

```diff
@@ -535,6 +530,22 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         }
         let o_creat = this.eval_libc_i32("O_CREAT")?;
         if flag & o_creat != 0 {
+            // Get the mode.  On macOS, the argument type `mode_t` is actually `u16`, but
+            // C integer promotion rules mean that on the ABI level, it gets passed as `u32`
+            // (see https://github.com/rust-lang/rust/issues/71915).
+            let mode = if let Some(arg) = args.get(2) {
+                this.read_scalar(arg)?.to_i32()?
```

Comment:
This was `to_u32` before, is there a reason you changed this?

Reply:
That seems like a typo, I don't know when that happened, good catch!

Reply:
You force-pushed, but the code is unchanged?

Reply:
Huh, I thought that git was meant to yell at you if you make an empty commit, I forgot to actually stage the changes :sweat_smile: Fixed